### ISSUE TITLE
Lab scheduling

### ISSUE THREAD

**Issue by thatoddmailbox** [PlanHubMe/PlanHub#9]
It could integrate with Labby or be something standalone (with an API).

This could then work with #1 and show you when you have labs.

**Comment by medensqui** [comment]
@thatoddmailbox Yes that was an idea I had, and during free labs you could add HW to do

**Comment by amazansky** [comment]
So here is my idea
In our login privacy statement it says "It's only used to sign you in and download your schedule and homework."
So...
- [x] we get schedule from schedules
- [ ] get a list of teachers that user has
- [ ] take away Lunch and Assembly teachers (etc)
- [ ] we have a list of teachers user has for classes
- [ ] remove duplicates like for house and math if you have the same teacher
- [ ] make a dropdown
  - [ ] if teacher has opted into lab scheduling, they are shown in green
  - [ ] else, they are un-clickable
  - [ ] at the bottom of the dropdown it would say something like "Want a teacher to sign up for Lab scheduling online? Refer them!" and the user would click something to trigger an email in Mandrill **Edit: or the SMTP and IMAP/POP3 server for planhub.me that @thatoddmailbox is hosting**.

Is that a lot of work on a scale of 1 - 10?

**Comment by willbarkoff** [comment]
@RobotXLabs 11

**Comment by thatoddmailbox** [comment]
@RobotXLabs that's doable and not really that hard. I'm not sure if it would be easier to use Schedules or Courses. Probably Schedules since we sign them in already.

**Comment by willbarkoff** [comment]
List of issues:
1. Mandrill costs money once you send 2 thousand emails. https://mandrill.com/pricing/
2. Schedules thinks it is 1889 (one of the issues we had with ULTIMATHEXRS/DaltonTab)
3. It is good practice to have at least 3 things in a list.

**Comment by thatoddmailbox** [comment]
@nywillb 
1. Who said we have to use mandrill? I'm already hosting a SMTP and IMAP/POP3 server for planhub.me emails, we can just use that.
2. So? We don't care about the dates, we care about the teacher. Even then, the times are valid. Just ignore the dates.
3. ...

**Comment by willbarkoff** [comment]
@thatoddmailbox 
1. We can also use [Amazon SES](https://aws.amazon.com/ses/?sc_channel=PS&sc_campaign=acquisition_US&sc_publisher=google&sc_medium=ses_b&sc_content=ses_e&sc_detail=amazon%20ses&sc_category=ses&sc_segment=73823394522&sc_matchtype=e&sc_country=US&s_kwcid=AL!4422!3!73823394522!e!!g!!amazon%20ses&ef_id=VfCeCgAAAJ@66oOp:20160102022344:s)
2. Good point
3. Still want three things.

**Comment by thatoddmailbox** [comment]
@nywillb 

> Email messages are charged at $0.10 per 1,000.

(from https://aws.amazon.com/ses/pricing/)

**Comment by willbarkoff** [comment]
@thatoddmailbox 

> AWS Free Tier offers 62,000 messages per month to any recipient when you call Amazon SES from an Amazon EC2 instance.

(from https://aws.amazon.com/ses/)

**Comment by thatoddmailbox** [comment]
@nywillb 

> The AWS Free Tier includes services with a free tier available for 12 months following your AWS sign-up date.

(from https://aws.amazon.com/free/)

**Comment by thatoddmailbox** [comment]
@RobotXLabs @nywillb Another option would be to store the Schedules session key and use http://rouxkeyextend.planhub.me/ to keep it alive when you want to make an actual request.

**Comment by amazansky** [comment]
That seems like a better idea to me.
